server/routes/journeys/update-referral/contributory-factor-comment/controller.ts

```diff
@@ -0,0 +1,42 @@
+import { NextFunction, Request, Response } from 'express'
+import { getMaxCharsAndThresholdForAppend, getTextForApiSubmission } from '../../../../utils/appendFieldUtils'
+import { UpdateReferralContributoryFactorController } from '../../base/updateReferralContributoryFactorController'
+
+export class UpdateContributoryFactorsCommentController extends UpdateReferralContributoryFactorController {
+  GET = async (req: Request, res: Response) => {
+    const selectedCf = this.getSelectedCf(req)
+
+    if (!selectedCf) {
+      return res.notFound()
+    }
+
+    return res.render('update-referral/contributory-factor-comment/view', {
+      isUpdate: true,
+      recordUuid: req.journeyData.csipRecord!.recordUuid,
+      currentComment: selectedCf.comment || '',
+      factorDescription: selectedCf.factorType.description,
+      comment: res.locals.formResponses?.['comment'],
+      ...getMaxCharsAndThresholdForAppend(res.locals.user.displayName, selectedCf.comment),
+    })
+  }
+
+  checkSubmitToAPI = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    const selectedCf = this.getSelectedCf(req)
+
+    if (!selectedCf) {
+      return res.notFound()
+    }
```

Comment:
this.updateContributoryFactor also has the null check and notFound handling inside. one of them can be removed?

Reply:
We're accessing this variable in order to call this.updateContributoryFactor so we can't remove this check